pkg/cmd/chat/events/events.go

```diff
@@ -0,0 +1,149 @@
+		`),
+		Example: heredoc.Doc(`
+			# Listen to events for user with id 'my-user-1'
+			$ stream-cli chat listen-events --user-id my-user-1
+
+			# Listen to events for user with id 'my-user-2' and keeping the connection open for 120 seconds
+			$ stream-cli chat listen-events --user-id my-user-1 --timeout 120
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			timeout, _ := cmd.Flags().GetInt32("timeout")
+			if timeout > 300 {
+				return fmt.Errorf("timeout cannot be greater than 300")
+			}
+			userID, _ := cmd.Flags().GetString("user-id")
+			config := config.GetConfig(cmd)
+
+			app, err := config.GetDefaultAppOrExplicit(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := config.GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			token, err := client.CreateToken(userID, time.Time{})
+			if err != nil {
+				return err
+			}
+
+			apiKey, _, err := config.GetCredentials(cmd)
+			if err != nil {
+				return err
+			}
+
+			cmd.Printf("> 🚨 Warning! The WebSocket connection can be expensive so we close it after %d seconds.\n", timeout)
+			time.Sleep(2 * time.Second)
+			// Giving the user 2 seconds to read the warning message
+			// because the first heartbeat is sent super quickly and
+			// takes up the whole screen.
+
+			url := getUrl(app, userID, apiKey, token)
+			websocket.DefaultDialer.HandshakeTimeout = time.Second * 5
+			conn, _, err := websocket.DefaultDialer.Dial(url, nil)
+			if err != nil {
+				return err
+			}
+			cmd.Println("> Successfully connected. Waiting for events...⌛️")
+
+			exit := make(chan os.Signal, 1)
+			signal.Notify(exit, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
+
+			// Since keeping connections can be expensive
+			// let's just exit after 'timeout' seconds.
+			go func() {
+				time.Sleep(time.Duration(timeout-10) * time.Second)
+				cmd.Println("> Exiting in 10 seconds...")
+				time.Sleep(10 * time.Second)
+				cmd.Printf("> %d seconds passed. Exiting now.\n", timeout)
+				exit <- syscall.SIGINT
+			}()
+
+			go func() {
+				for {
+					var event map[string]any
+					err := conn.ReadJSON(&event)
+					if err != nil {
+						cmd.PrintErr(err)
+						continue
+					}
+
+					err = utils.PrintObject(cmd, event)
+					if err != nil {
+						cmd.PrintErr(err)
+						continue
+					}
+
+					cmd.Println("> Press Ctrl+C to exit")
+				}
+			}()
+
+			<-exit
+			cmd.Println("> Exiting")
+			return conn.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.Int32P("timeout", "t", 60, "[optional] For how many seconds do we keep the connection alive. Default is 60 seconds, max is 300.")
+	fl.StringP("user-id", "u", "", "[required] User ID")
+	fl.StringP("output-format", "o", "json", "[optional] Output format")
+	_ = cmd.MarkFlagRequired("user-id")
+
+	return cmd
+}
+
+func getUrl(app *config.App, userID, apiKey, token string) string {
+	u, _ := url.Parse(app.ChatURL)
```

Comment:
would be better not to skip errors in general, it might fail somewhere else and then error message could be harder to understand

Reply:
✅